src/ChannelWrapper.coffee

```diff
@@ -131,20 +160,22 @@ class ChannelWrapper extends EventEmitter
     # Returns a Promise which resolves when this channel next connects.
     # (Mainly here for unit testing...)
     waitForConnect: pb.break ->
-        if @_channel
+        if @_channel and !@_settingUp
             return Promise.resolve()
         else
             return new Promise (resolve) => @once 'connect', resolve
 
+    _shouldPublish: -> (@_messages.length > 0) and !@_settingUp and @_channel
```

Comment:
Was going to suggest maybe setting `@_settingUp` to `Promise.resolve()` when it has nothing to do, but this is a synchronous check for `null`, so you probably don't want to do that.